vivisect/tests/teststorage.py

```diff
@@ -124,3 +125,28 @@ def test_bad_event(self):
         self.assertIn("IndexError: list index out of range", ''.join(logcap.output))
         self.assertEqual(1, len(files))
         self.assertEqual('VivisectFile', files[0])
+
+    def test_basicfile_header(self):
+        basicfile = tempfile.NamedTemporaryFile(delete=False)
+        vw = vivisect.VivWorkspace()
+        add_events(vw)
+        vw.setMeta('StorageName', basicfile.name)
+        vw.saveWorkspace(False)
+
+        # test the header was added correctly
+        with open(basicfile.name, 'rb') as f:
+            header = f.read(8)
+            self.assertEqual(header, vsbf.vivsig_cpickle)
+
+        # now add events and make sure we don't add the header unnecessarily
+        vw = vivisect.VivWorkspace()
+        vw.loadWorkspace(basicfile.name)
+        vw.addLocation(0x6100, 3, 5)
+        vw.addLocation(0x6105, 3, 5)
+        vw.addLocation(0x610a, 3, 5)
+        vw.saveWorkspace(False)
+
+        # now load it.  if a VIV\0\0\0\0\0 was added incorrectly, this will blow up
+        vw = vivisect.VivWorkspace()
+        vw.loadWorkspace(basicfile.name)
```

Comment:
Mostly to assuage my own paranoia about things breaking in the future, could you add a few asserts here asserting on locations made between the first and second phases like the 0x610a location?

I'm sure it works. But having that nailed down and encoded would be nice :)